lib/interactive_viewer.ml

```diff
@@ -111,6 +121,114 @@ let change_summary z_patches : ui Lwd.t =
   in
   W.string ~attr:Notty.A.(fg lightcyan) operation_count
 
+(** Side by side diff view implementation **)
+
+type view_mode = SideBySide | Normal
+
+let view_mode = Lwd.var Normal
+
+let toggle_view_mode () =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let rec split_and_align_hunk hunks mine_acc their_acc =
+  match hunks with
+  | [] -> (List.rev mine_acc, List.rev their_acc)
+  | (`Common _ as common) :: t ->
+      split_and_align_hunk t (common :: mine_acc) (common :: their_acc)
+  | `Mine s :: `Their s' :: t ->
+      split_and_align_hunk t (`Mine s :: mine_acc) (`Their s' :: their_acc)
+  | `Mine s :: t ->
+      split_and_align_hunk t (`Mine s :: mine_acc) (`Common "" :: their_acc)
+  | `Their s :: t ->
+      split_and_align_hunk t (`Common "" :: mine_acc) (`Their s :: their_acc)
+
+let wrap_text s =
+  let rec aux acc line =
+    if String.length line <= 0 then List.rev acc
+    else
+      let before, after =
+        if String.length line <= 40 then (line, "")
+        else
+          let split_index =
+            try String.rindex_from line 40 ' ' with Not_found -> 40
+          in
+          ( String.sub line 0 split_index,
+            String.sub line (split_index + 1)
+              (String.length line - split_index - 1) )
+      in
+      aux (before :: acc) after
+  in
+  aux [] s
+
+let max_segment_width hunks =
+  let rec aux max_width lines =
+    match lines with
+    | [] -> max_width
+    | (`Common s | `Mine s | `Their s) :: rest ->
+        let width = String.length s in
+        aux (max max_width width) rest
+  in
+  List.fold_left
+    (fun acc hunk ->
+      let mine_lines, their_lines =
+        split_and_align_hunk hunk.Patch.lines [] []
+      in
+      let lines = List.concat [ mine_lines; their_lines ] in
+      aux acc lines)
+    0 hunks
+
+let lines_to_ui lines max_width attr_line_number attr_change =
+  List.flatten
+    (List.map
+       (fun line ->
+         let content, attr =
+           match line with
+           | `Common s -> (s, attr_line_number)
+           | `Mine s -> (s, attr_change)
+           | `Their s -> (s, attr_change)
+         in
+         let padded_content =
+           content ^ String.make (max_width - String.length content) ' '
```

Comment:
What is the goal of this padding ?

It's currently misbehaving and adding a variable amount of unwanted empty lines between diff lines. Though, I wonder if there's a way not to use padding.

Reply:
I added padding initially as an attempt to make the spacing look better when text wrapping.. but since we want to forgo text wrapping at this moment I removed this (saved for later) as I don't think I'll be needing this without the text wrap function